homeassistant/components/cert_expiry/sensor.py

```diff
@@ -102,50 +131,42 @@ def available(self):
         """Return the availability of the sensor."""
         return self._available
 
-    async def async_added_to_hass(self):
-        """Once the entity is added we should update to get the initial data loaded."""
-
-        @callback
-        def do_update(_):
-            """Run the update method when the start event was fired."""
-            self.async_schedule_update_ha_state(True)
-
-        if self.hass.is_running:
-            self.async_schedule_update_ha_state(True)
-        else:
-            # Delay until HA is fully started in case we're checking our own cert.
-            self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, do_update)
-
-    def update(self):
+    async def async_update(self):
         """Fetch the certificate information."""
         try:
-            cert = get_cert(self.server_name, self.server_port)
-        except socket.gaierror:
-            _LOGGER.error("Cannot resolve hostname: %s", self.server_name)
-            self._available = False
-            self._valid = False
-            return
-        except socket.timeout:
-            _LOGGER.error("Connection timeout with server: %s", self.server_name)
+            days_to_expiry = await get_cert_time_to_expiry(
+                self.hass, self.server_name, self.server_port
+            )
+        except TemporaryFailure as err:
+            _LOGGER.error(err)
             self._available = False
+            self._error = err
             self._valid = False
             return
-        except (ssl.CertificateError, ssl.SSLError):
+        except ValidationFailure as err:
+            _LOGGER.error(
+                "Certificate validation error: %s [%s]", self.server_name, err
+            )
             self._available = True
+            self._error = err
             self._state = 0
             self._valid = False
             return
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error checking %s:%s", self.server_name, self.server_port
+            )
+            self._available = False
+            self._error = "Unknown"
+            self._valid = False
```

Comment:
As above. We don't know if it's valid or unvalid, so it's better to not change it.